crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -566,6 +616,155 @@ impl Object {
 
         Ok(conn)
     }
+
+    async fn query_live(db: &Db, address: SuiAddress) -> Result<Option<Self>, Error> {
+        let vec_address = address.into_vec();
+        use objects::dsl as objects;
+
+        let stored_obj: Option<StoredObject> = db
+            .execute(move |conn| {
+                conn.result(move || {
+                    objects::objects
+                        .filter(objects::object_id.eq(vec_address.clone()))
+                        .limit(1)
+                        .into_boxed()
+                })
+                .optional()
+            })
+            .await
+            .map_err(|e| Error::Internal(format!("Failed to fetch object: {e}")))?;
+
+        stored_obj.map(Self::try_from).transpose()
+    }
+
+    async fn query_historical(
+        db: &Db,
+        address: SuiAddress,
+        version: Option<u64>,
+        checkpoint_sequence_number: Option<u64>,
+    ) -> Result<Option<Self>, Error> {
+        let vec_address = address.into_vec();
+        let version = version.map(|n| n as i64);
+        let checkpoint_sequence_number = checkpoint_sequence_number.map(|n| n as i64);
+
+        use checkpoints::dsl as checkpoints;
+        use objects_history::dsl as history;
+        use objects_snapshot::dsl as snapshot;
+
+        let results: Option<Vec<StoredHistoryObject>> = db
+            .execute(move |conn| {
+                conn.results(move || {
+                    // If an object was created or mutated in a checkpoint outside the current
+                    // available range, and never touched again, it will not show up in the
+                    // objects_history table. Thus, we always need to check the objects_snapshot
+                    // table as well.
+                    let mut snapshot_query = snapshot::objects_snapshot
+                        .filter(snapshot::object_id.eq(vec_address.clone()))
```

Comment:
nit: Prefer not creating `vec_address` and then having to clone it here -- just call `address.into_vec()`  in situ -- `SuiAddress` has `Copy` so it's fine to do this.

Reply:
strange .. i swear i got a rust complaint about using a moved value. but yes, address.into_vec() is fine to do now, thanks